docs/ape17.rst

```diff
@@ -0,0 +1,533 @@
+Astropy package template, it is likely that you have functions called
+``get_package_data`` defined inside ``setup_package.py`` files. Remove
+these functions, and instead define the package data using a
+``[options.package_data]`` section inside your ``setup.cfg`` file,
+e.g.::
+
+   [options.package_data]
+   * = *.fits, *.csv
+   my_package.tests = data/*
+
+In the above example, all ``.fits`` and ``.csv`` in the package will be
+included as well as all files inside ``my_package/tests/data``.
+
+Step 5 - Update your ``setup.py`` file
+--------------------------------------
+
+Copy the ``setup.py`` file you generated in :ref:`Step 0 <step0>` and replace your existing one
+- it should be good to go as-is without any further customizations.
+
+Step 6: add a ``pyproject.toml`` file
+-------------------------------------
+
+The ``pyproject.toml`` file is used to declare dependencies needed to run
+``setup.py`` and build the package. Copy the ``pyproject.toml`` file you
+generated in :ref:`Step 0 <step0>` and replace your existing one.
+
+If your package doesn’t have any compiled extensions, the file should contain:
+
+.. code:: toml
+
+   [build-system]
+   requires = ["setuptools",
+               "setuptools_scm",
+               "wheel"]
+   build-backend = 'setuptools.build_meta'
+
+Step 7 - Handling C/Cython extensions
+-------------------------------------
+
+If your package has no compiled C/Cython extensions, you can skip this
+step. Otherwise, if you have C or Cython extensions, you can either
+define your extensions manually inside the ``setup.py`` file or make use
+of the `extension-helpers <https://extension-helpers.readthedocs.io>`__
+package to collect extensions in a similar way to astropy-helpers.
+
+Step 7a - Defining extensions manually
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can define extensions manually as described
+`here <https://oa-packaging-guide-preview.readthedocs.io/en/latest/extensions.html#defining-extensions-in-setup-py>`__.
+If you do this, you can remove all ``setup_package.py`` files in your
+package, and you don't need to include extension-helpers in the
+``pyproject.toml`` file.
+
+If you have Cython extensions or your extensions use the NumPy C API,
+proceed to Step 7c, otherwise you can proceed to Step 8.
+
+Step 7b - Using extension-helpers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the extension-helpers package to:
+
+-  Automatically define extensions for Cython files
+-  Pick up extensions declared in ``setup_package.py`` files, as
+   described in the `extension-helpers
+   documentation <https://extension-helpers.readthedocs.io/en/latest/>`__.
+
+The latter works by looking through all the ``setup_package.py`` files
+in your package and executing the ``get_extensions()`` functions, which
+each should return a list of extensions. Check through your existing
+``setup_package.py`` files (if any), and make sure that any
+``astropy_helpers`` imports are changed to ``extension_helpers``.
```

Comment:
If I remember correctly, a simple search and replace is not enough. You also moved some API up.

There is also a change from string to function call that has something to do with Numpy `get_include`.

Reply:
Fixed!